### PR TITLE
Generate constructor from vertical member selection

### LINKED ISSUE #28774: Selection to trigger 'Generate constructor' is finicky
VS 15.8 Preview 4

When selecting lines, I always use vertical motions of the mouse since it's easier and takes the previous newline with it which is 99% of the time what I want. If I'm typing, I'll use <kbd>Shift</kbd>+<kbd>↑</kbd> for each line I want to select which does the same thing.

However, this type of selection never triggers the expected code actions. This is what it looks like. First I do the selection that I'm used to, and no refactoring icon shows up (<kbd>Ctrl</kbd>+<kbd>.</kbd> does nothing, which is what I actually use). Then, I do the slower selection to demonstrate the refactoring options that I'd expect to appear the first time:

sensitive selection

### PR DESCRIPTION
**Pull request by jnm2**
Fixes #28774.

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateConstructorFromMembers/GenerateConstructorFromMembersTests.cs [hunk 1]
```diff
@@ -136,6 +136,45 @@ class Z
...
+        public async Task TestMultipleFields_SelectionIncludingClassOpeningBrace()
+        {
+            await TestMissingAsync(
+@"using System.Collections.Generic;
+
+class Z
+[|{
+    int a;
+    string b;|]
+}");
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateConstructorFromMembers/GenerateConstructorFromMembersTests.cs]: let's have a test where the selection goes up to a preceding member, not just to the open brace of the file.  one test with a field that is higher, and one with a method would be sufficient for me.

File: src/EditorFeatures/VisualBasicTest/GenerateConstructorFromMembers/GenerateConstructorFromMembersTests.vb [hunk 2]
```diff
@@ -46,6 +46,33 @@ End Class",
...
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateConstructorFromMembers)>
+        Public Async Function TestMultipleFields_SelectionIncludingClassName() As Task
+            Await TestMissingAsync(
+"Class Progra[|m
+    Private i As Integer
+    Private b As String|]
+End Class")
+        End Function
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/GenerateConstructorFromMembers/GenerateConstructorFromMembersTests.vb]: can we have a test where the class has an "inherits" clause on the next line.  also one where the class is generic.  also one where there is a preceding function and a preceding field.

also test if we exactly select the fields and no whitespace. i.e.:

```vb
+    [|Private i As Integer
+    Private b As String|]
```

File: src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateConstructorFromMembers/GenerateConstructorFromMembersTests.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateConstructorFromMembers/GenerateConstructorFromMembersTests.cs]: also a test where you exactly grab both fields.  i.e.:

```c#
+    [|int a;
+    string b;|]
```

Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharpTest/GenerateFromMembers/GenerateConstructorFromMembers/GenerateConstructorFromMembersTests.cs]: 👍

**Comment by jnm2** [comment]
@CyrusNajmabadi Added.

**Pull request merged**